### PR TITLE
Manifest-v3 Functions

### PR DESCRIPTION
**Pull request by ghost**
<details> <summary>NOTICE</summary>
I dedicate any and all copyright interest in this software to the
public domain. I make this dedication for the benefit of the public at
large and to the detriment of my heirs and successors. I intend this
dedication to be an overt act of relinquishment in perpetuity of all
present and future rights to this software under copyright law.
</details>

Fixes: Link to issue

I transferred functions from the mv2 injection script to the mv3 helpers script.

- [x] I made sure there are no unnecessary changes in the code*
- [ ] Tested on Chromium- Browser OS
- [ ] Tested on Firefox

\* indicates required

### CODE REVIEW

File: src/helpers/dom.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
+//Constants needed for bypassing
+const ODP=(t,p,o)=>{try{Object.defineProperty(t,p,o)}catch(e){console.trace("[FastForward] Couldn't define",p)}}
```
Comment (reashetyrr) [inline src/helpers/dom.js]: I'd prefer if you also export them, then we can do `import { ODP, showInfoBox } from 'helpers'`

File: src/helpers/dom.js [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+const setTimeout=window.setTimeout,setInterval=window.setInterval,URL=window.URL,docSetAttribute=document.documentElement.setAttribute.bind(document.documentElement)
```
Comment (reashetyrr) [inline src/helpers/dom.js]: It's prettier and cleaner to just declare the constants on new lines

File: src/helpers/dom.js [hunk 3]
```diff
@@ -24,7 +28,88 @@ export function insertInfoBox(text) {
...
+}
+export function crowdContribute(target,f){
+    if(typeof f!= "function"){
+        f=() => {}
+    }
+    if(crowdEnabled && isGoodLink(target)){
+        docSetAttribute("{{channel.crowd_contribute}}",target)
+        setTimeout(f,10)
+    }
+    else{
```
Comment (reashetyrr) [inline src/helpers/dom.js]: We might want to look into writing the variables in full names: `f => func` etc so we can clearly see what the variable should contain

**Review by reashetyrr — COMMENTED** [review]
Please check my comments ☺️

File: src/helpers/dom.js (hunk 2 above)
Comment (ghost) [reply src/helpers/dom.js]: Competitive programming moment 💀

**Comment by ghost** [comment]
@reashetyrr I've fixed the issues raised and it should meet what you described earlier.

File: src/helpers/dom.js [hunk 4]
```diff
@@ -1,105 +1,196 @@
...
+            clearInterval(time);
+        }
+    }, 10);
+    setInterval(() => {
+        clearInterval(time);
+    }, 30000);
+}
+export function crowdDomain(domain) {
+    if (crowdEnabled && domain) {
+        docSetAttribute("{{channel.crowd_domain}}", domain);
```
Comment (reashetyrr) [inline src/helpers/dom.js]: Where does the crowdenabled come from?

File: src/helpers/dom.js [hunk 5]
```diff
@@ -1,105 +1,196 @@
...
+    if (remains) {
+        docSetAttribute("{{channel.crowd_referer}}", query);
+    }
+}
+export function crowdBypass(func, available) {
+    if (!func) {
+        func = () => {};
+    }
+    if (crowdEnabled) {
+        if (ignoreCrowdBypass) {
```
Comment (reashetyrr) [inline src/helpers/dom.js]: Where does ignorecrowdbypass come from

File: src/helpers/dom.js [hunk 6]
```diff
@@ -1,105 +1,196 @@
...
+        docSetAttribute("{{channel.crowd_domain}}", domain);
+    }
+}
+export function crowdPath(path) {
+    if (crowdEnabled && path) {
+        docSetAttribute("{{channel.crowd_path}}", path);
+    }
+}
+export function crowdReferer(query) {
+    if (remains) {
```
Comment (reashetyrr) [inline src/helpers/dom.js]: Where does rains come from

File: src/helpers/dom.js [hunk 7]
```diff
@@ -1,105 +1,196 @@
...
+    }
+    if (crowdEnabled && isGoodLink(target)) {
+        docSetAttribute("{{channel.crowd_contribute}}", target);
+        setTimeout(func, 10);
+    } else {
+        func();
+    }
+}
+export function contributeAndNavigate(target) {
+    if (!navigated && isGoodLink(target)) {
```
Comment (reashetyrr) [inline src/helpers/dom.js]: Where does navigated come from

File: src/helpers/dom.js [hunk 8]
```diff
@@ -1,116 +1,256 @@
...
+  crowdBypass,
+  crowdContribute,
+  contributeAndNavigate,
+  unsafelyNavigate,
+  parseTarget,
+  safelyNavigate,
+  unsafelyAssign,
+  safelyAssign,
+  isGoodLink,
+  bypassRequests,
```
Comment (reashetyrr) [inline src/helpers/dom.js:255]: you might want to also put ODP here

**Review by reashetyrr — APPROVED** [review]

**Pull request merged**